lib/ssl-helper.js

```diff
@@ -13,4 +15,46 @@ function upgradeToSSL (socket, key, cert) {
     })
   });
 }
-export { upgradeToSSL };
+
+/**
+ * after the ssl protocol is successfully handshake, close the ssl protocol channel and use text transmission
+ * @param socket
+ * @param key
+ * @param cert
+ * @returns {Promise<*>}
+ */
+async function enableSSLHandshakeOnly (socket, key, cert) {
+  const sslSocket = tls.connect({
+    socket,
+    secureContext: tls.createSecureContext({
+      key,
+      cert,
+      secureProtocol: TLS_VERSION
+    }),
+    rejectUnauthorized: false,
+  });
+  let isSecureConnect = false;
+
+  // stop receiving data after successful handshake
+  sslSocket.on('secureConnect', () => {
+    sslSocket._handle.readStop();
+    isSecureConnect = true;
+  });
+
+  await waitForCondition(() => isSecureConnect, {
```

Comment:
probably for this particular case the usage is not optimal, because it would still make 500ms delay after the connection is done. I wuld try something like that:

```
await new B((resolve, reject) => {
  sslSocket.once('secureConnect', () => {
    return resolve();
  });
  
  setTimeout(HANDSHAKE_TIMEOUT, () => {
      if (!sslSocket.destroyed) {
       sslSocket.end();
     }
     return reject(new Error('ssl handshake error'));
  });
});

return net.Socket({fd: socket._handle.fd});
```